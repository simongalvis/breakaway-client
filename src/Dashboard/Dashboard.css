#expandedactivitylink{
    color:black;
}
.Dashboard{
    min-height: 100%;
}