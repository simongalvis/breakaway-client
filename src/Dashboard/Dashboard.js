import React from 'react'
import { Link } from 'react-router-dom'
import './Dashboard.css'
import DashboardNavBar from '../DashboardNavBar/DashboardNavBar'
import Activity from '../Activity/Activity'
import {activityStore} from '../store'
import ApiContext from '../ApiContext'
import activity from '../Activity/Activity'





class Dashboard extends React.Component{

    static contextType = ApiContext

    activitiesToDisplay = this.context.activities.filter(activity => activity.user_id == this.context.loggedUser.id)
    user = activity => this.context.users.find(user => activity.user_id == user.id)

    render(){
        return(
            <div className='Dashboard'>
                <DashboardNavBar/>
                <h2>Activities</h2>
                {this.activitiesToDisplay.length > 0 ?
                this.activitiesToDisplay.map(activity => (
                    <Activity key={activity.id}
                              name={this.user(activity).fullname}
                              title={activity.title}
                              duration={activity.duration} 
                              distance={activity.distance}
                              description={activity.description}
                              date={activity.date_published}/>
                ))
                : <h5>No activities recorded yet! To get started, add your first activity by clicking the blue "Breakaway" button on the top right.</h5>}
                    
                
            </div>
            
        )
    }
}


export default Dashboard;