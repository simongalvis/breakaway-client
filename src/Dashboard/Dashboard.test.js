import React from 'react'
import ReactDOM from 'react-dom'
import Dashboard from './Dashboard'
import renderer from 'react-test-renderer';

describe(`Dashboard component`, () =>{
  it(`renders without crashing`, () =>{
    const div = document.createElement('div');
  
    ReactDOM.render(<Dashboard/>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  })
  it(`Renders UI as expected`, () =>{
    const tree = renderer 
      .create(<Dashboard/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
