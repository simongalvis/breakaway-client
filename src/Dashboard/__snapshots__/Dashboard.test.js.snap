// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Renders UI as expected 1`] = `
<div
  className="Dashboard"
>
  <header>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="Blue breakaway logo"
        id="main-logo"
        src="breakaway-logo.png"
      />
    </a>
    <a
      aria-current={null}
      href="/new-activity"
      onClick={[Function]}
    >
      <button
        className="nav-button"
        id="newactivity-breakaway-button"
        value="new activity"
      >
        <b>
           +Breakaway!
        </b>
          
      </button>
    </a>
  </header>
  <h2>
    Activities
  </h2>
  <h5>
    No activities recorded yet! To get started, add your first activity by clicking the blue "Breakaway" button on the top right.
  </h5>
</div>
`;
